### PR TITLE
first time submission

### PR DESCRIPTION
**Pull request by meta-manoj-sharma**

### CODE REVIEW

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/Main.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+						employeeName = input.nextLine();
+						if(employeeName.charAt(0)==' ')
+						{
+							throw new Exception();
+						}
+						System.out.println("enter Id of employee");
+						employeeId = input.nextInt();
+						System.out.println("enter Age. of employee");
+						employeeAge = input.nextInt();
+						if(employeeAge >= 18){
```
Comment (meta-Narendra-Agrawal) [inline DSA-Session-6/EmployeeQuickSort/src/quicksort/Main.java]: curly braces should always be in new line for better readibility.

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	Node headNode;// For storing head position
+
+	/**
+	 * This method helps in inserting any new nodes to the list
+	 * @param list is the linked list containing head
+	 * @param data is the employee object to be inserted in list
+	 * @return the linked list with its head position
+	 */
+	public LinkedListOperation addEmployee(LinkedListOperation list, Employee data) {
+		Node newNode = new Node(data);
```
Comment (meta-Narendra-Agrawal) [inline DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java]: Here you are using data but checking null in below lines. Check null first and then proceed if data is not null.

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+		try {
```
Comment (meta-Narendra-Agrawal) [inline DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java]: Remove try catch and write whole logic inside this:
if(data != null)
{
//your logic
}
else
throw error

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperationTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class LinkedListOperationTest {
+	LinkedListOperation object=new LinkedListOperation();
+	@Test
+	public void testSortList1() throws Exception 
+	{	
+			LinkedListOperation objectOfEmployeeList=new LinkedListOperation();
```
Comment (meta-Narendra-Agrawal) [inline DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperationTest.java]: write two more test methods for these scenarios:
 1. two or more employees having Same salary but different age.
2. two or more employees having same age and same salary.

File: DSA-Session-6/WeightedGraph/src/graph/GraphTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import org.junit.Test;
+
+public class GraphTest {
+
+	/**
+	 * test case for is connected
+	 * @throws Exception
+	 */
+	@Test
+	public void isConnectedTest() throws Exception
```
Comment (meta-Narendra-Agrawal) [inline DSA-Session-6/WeightedGraph/src/graph/GraphTest.java]: Why test method is throwing exception? Write one more test method to cover the scenario where graph is not connected.

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/Main.java (hunk 1 above)
Comment (meta-manoj-sharma) [reply DSA-Session-6/EmployeeQuickSort/src/quicksort/Main.java]: done the changes

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java (hunk 2 above)
Comment (meta-manoj-sharma) [reply DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java]: logic updated

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java (hunk 3 above)
Comment (meta-manoj-sharma) [reply DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperation.java]: logic updated

File: DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperationTest.java (hunk 4 above)
Comment (meta-manoj-sharma) [reply DSA-Session-6/EmployeeQuickSort/src/quicksort/LinkedListOperationTest.java]: test cases added

File: DSA-Session-6/WeightedGraph/src/graph/GraphTest.java (hunk 5 above)
Comment (meta-manoj-sharma) [reply DSA-Session-6/WeightedGraph/src/graph/GraphTest.java]: test cases added

**Pull request merged**